feat: track awesome notebooks created over time

SanjuEpic opened the pull request:
Referring to issue #302

@jravenel

FlorentLvr commented:
@SanjuEpic , TY for your contribution !
I am not sure the code is correct regarding the graph. 
Using the commit ID is a good way but you should :
- Use then this url => https://api.github.com/repos/jupyter-naas/awesome-notebooks/git/trees/bbff5b56d8c0f6821088536a8b90e2c43c74473e to get the tree detailed
- Inside the object "tree" use the url to get the detailed.
- Rebuild the repository image for each commits
- Count number of notebook for all commits
Adding the formula get_commit from your other contribution will help you.
Hope my comments will help.
Don't hesitate if you have any question.
![image](https://user-images.githubusercontent.com/48032461/151407487-e8d21869-f243-4b6d-9299-8e1b732f4563.png)

SanjuEpic commented:
yep @FlorentLvr my commited code is outdated, made some changes to this code, but still working on it, to update this PR.
Thanks!

FlorentLvr commented:
@SanjuEpic, no problem. Just push your commits when it's done and I will have a look at it.

SanjuEpic commented:
additional comments related to gitguardian checks:
the GitHub token will expire today.

FlorentLvr commented:
Hi @SanjuEpic ! TY for your contribution.
I just checked your notebook, it is working well and the code is well structured. Well done 👍 
However, the total number of notebook is wrong. You should have 230 instead of 309. 
Could you try to fix it ?
The total number should match with len of template.json. This json is generated everytime we merge a PR.
TY 

![image](https://user-images.githubusercontent.com/48032461/151658753-cca245df-0b34-4751-8d02-997b901aff4d.png)

SanjuEpic commented:
Yeah @FlorentLvr , tried many methods of deducing the right number of notebooks, but there isn't sufficient information to deduce this.
There could be some notebooks which might have got combinations of status like renamed, modified, removed and then added. These combinations somehow make this count increase to 309. 

Normally our algorithm take into consideration of only those commits for which the notebooks are either added or removed, and accordingly adding up the count of notebooks on that particular commit date. (+1 for added, -1 for removed)

Any thoughts would be helpful.
We can discuss on Monday regarding the same
Thanks!

FlorentLvr commented:
@SanjuEpic, I used the method I mentionned above to get notebooks by commits. I keep only the commits made by Github with message started by 'Merge'. The notebook take a bit of time but it's working !

@jravenel => notebook tracks by date
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

jravenel commented:
> @SanjuEpic, I used the method I mentionned above to get notebooks by commits. I keep only the commits made by Github with message started by 'Merge'. The notebook take a bit of time but it's working !
> 
> @jravenel => notebook tracks by date ![image](https://user-images.githubusercontent.com/48032461/152063660-7c26381e-de20-4a72-85fd-f343deb21ff3.png)

Awesome!

SanjuEpic commented:
Awesome one!
Great logic, keep only the commits made by Github with message started by 'Merge'.
Thanks for giving me this KT @FlorentLvr
